### PR TITLE
Added support for swift package manager

### PR DESCRIPTION
**Pull request by BarredEwe**
I have added full support for the swift package manager.

Now you can try this:
```swift
 .package(url: "https://github.com/BarredEwe/SwinjectStoryboard.git", .branch("swiftPM"))
 ```

### CODE REVIEW

**Comment by armintelker** [comment]
I like it :)

**Comment by igorkulman** [comment]
Nice idea, rewriting Objective C swizzling into Swift instead of fighting SPM.

File: Sources/Container+SwinjectStoryboard.swift [hunk 1]
```diff
@@ -6,6 +6,9 @@
 //  Copyright © 2015 Swinject Contributors. All rights reserved.
 //
 
+#if os(iOS) || os(tvOS)
```
Comment (BalazsSzamody) [inline Sources/Container+SwinjectStoryboard.swift]:
```suggestion
#if canImport(UIKit)
```

This is a bit better IMHO.

File: Sources/OSX/NSStoryboard+Swizzling.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#if os(OSX)
```
Comment (BalazsSzamody) [inline Sources/OSX/NSStoryboard+Swizzling.swift]:
```suggestion
#if canImport(Cocoa)
```

File: Sources/OSX/_SwinjectStoryboardBase(OSX).swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#if os(OSX)
```
Comment (BalazsSzamody) [inline Sources/OSX/_SwinjectStoryboardBase(OSX).swift]:
```suggestion
#if canImport(Cocoa)
```

File: Sources/SwinjectStoryboard+StoryboardReference.swift [hunk 4]
```diff
@@ -6,6 +6,13 @@
 //  Copyright © 2016 Swinject Contributors. All rights reserved.
 //
 
+import Foundation
+#if os(iOS) || os(tvOS)
```
Comment (BalazsSzamody) [inline Sources/SwinjectStoryboard+StoryboardReference.swift]:
```suggestion
#if canImport(UIKit)
```

File: Sources/SwinjectStoryboard+StoryboardReference.swift [hunk 5]
```diff
@@ -6,6 +6,13 @@
...
+    import UIKit
+#elseif os(OSX)
```
Comment (BalazsSzamody) [inline Sources/SwinjectStoryboard+StoryboardReference.swift]:
```suggestion
#elseif canImport(Cocoa)
```

**Comment by juliobailon** [comment]
When is this going to MASTER? Still Swift Package Manager does not work

**Comment by juliobailon** [comment]
In Xcode 12.3 it seems that the SwinjectStoryboard.configure() is doing nothing.

**Comment by BarredEwe** [comment]
> @BarredEwe In Xcode 12.3 it seems that the SwinjectStoryboard.configure() is doing nothing.

I fixed it into latest commit.

**Comment by BarredEwe** [comment]
After the last commit, you don't need to use 'SwinjectStoryboard.configure()'. Everything works as usual.

**Comment by ChediB** [comment]
@mpdifran

**Comment by mpdifran** [comment]
I'll take a look at getting this in, would be great to have SPM support!

File: Sources/SwinjectStoryboard.swift [hunk 6]
```diff
@@ -7,6 +7,11 @@
 //
 
 import Swinject
+#if os(iOS) || os(tvOS)
```
Comment (mpdifran) [inline Sources/SwinjectStoryboard.swift:10]:
```suggestion
#if canImport(UIKit)
```

File: Sources/SwinjectStoryboard.swift [hunk 7]
```diff
@@ -7,6 +7,11 @@
...
+    import UIKit
+#elseif os(OSX)
```
Comment (mpdifran) [inline Sources/SwinjectStoryboard.swift:12]:
```suggestion
#elseif canImport(OSX)
```

File: Sources/ViewController+Swinject.swift [hunk 8]
```diff
@@ -9,6 +9,7 @@
 import ObjectiveC
 
 #if os(iOS) || os(tvOS)
```
Comment (mpdifran) [inline Sources/ViewController+Swinject.swift:11]:
```suggestion
#if canImport(UIKit)
```

File: Sources/ViewController+Swinject.swift [hunk 9]
```diff
@@ -26,6 +27,7 @@ extension UIViewController: RegistrationNameAssociatable, InjectionVerifiable {
 }
 
 #elseif os(OSX)
```
Comment (mpdifran) [inline Sources/ViewController+Swinject.swift:29]:
```suggestion
#elseif canImport(Cocoa)
```

File: Sources/iOS-tvOS/UIStoryboard+Swizzling.swift [hunk 10]
```diff
@@ -0,0 +1,41 @@
+#if os(iOS) || os(tvOS)
```
Comment (mpdifran) [inline Sources/iOS-tvOS/UIStoryboard+Swizzling.swift:1]:
```suggestion
#if canImport(UIKit)
```

File: Sources/iOS-tvOS/_SwinjectStoryboardBase(iOS-tvOS).swift [hunk 11]
```diff
@@ -0,0 +1,27 @@
+#if os(iOS) || os(tvOS)
```
Comment (mpdifran) [inline Sources/iOS-tvOS/_SwinjectStoryboardBase(iOS-tvOS).swift:1]:
```suggestion
#if canImport(UIKit)
```

**Pull request merged**